### ISSUE TITLE
calcs

### ISSUE THREAD

**Issue by hammad93** [apatel726/HurricaneDissertation#68]
There isn't a way to calculate the difference between two trajectories or intensities

**Comment by hammad93** [comment]
line 78 and 79 represent some of the calcs. we need to find a better way to compare lat and lon instead of subtracting them

**Comment by hammad93** [comment]
https://github.com/apatel726/HurricaneDissertation/blob/master/test.py#L79

**Comment by hammad93** [comment]
https://www.nhc.noaa.gov/nhcexit.php?outurl=http://edwilliams.org/gccalc.htm
http://edwilliams.org/gccalc.htm

**Comment by hammad93** [comment]
is there a pure python way of doing this without http

**Comment by hammad93** [comment]
https://github.com/keras-team/keras/blob/v2.10.0/keras/losses.py#L308
we need to edit the error somehow because the loss is just `y_true - y_pred`

**Comment by hammad93** [comment]
The NHC formula can be used to match existing research standards. The calculator on their website utilizes https://www.nhc.noaa.gov/gccalc2.js, but it is unclear what the usage license is. The NHC credits the Great Circle Calculator (http://edwilliams.org/avform147.htm) work for their own formulae. There are available Python libraries with clear licenses that can be integrated based on the Great Circle Calculator.

An alternative can be to recreate the formulae in our own code. However, it may lead to additional delays, and we can always come back to it. In addition, the calculators will likely be valuable for custom-loss functions, and visualizations, among other use cases that still need to be fully understood but will take extra development compared to an existing library.

**Comment by hammad93** [comment]
Reference https://github.com/apatel726/HurricaneDissertation/commit/3349dc49d0f2e115f275fd6635e933da821b406c for integration of great circle calculations.
